megamek/src/megamek/common/Mounted.java

```diff
@@ -1694,18 +1694,68 @@ public void setModeSwitchable(boolean b) {
     
     /**
      * Method that checks to see if our capital missile bay is in bearings-only mode
+     * Only available in space games
      * @return
      */
     public boolean isInBearingsOnlyMode() {
-        if (curMode().equals("Bearings-Only Extreme Detection Range")
+        if ((curMode().equals("Bearings-Only Extreme Detection Range")
                     || curMode().equals("Bearings-Only Long Detection Range")
                     || curMode().equals("Bearings-Only Medium Detection Range")
-                    || curMode().equals("Bearings-Only Short Detection Range")) {
+                    || curMode().equals("Bearings-Only Short Detection Range")
+                    || curMode().equals("Waypoint Launch Bearings-Only Extreme Detection Range")
+                    || curMode().equals("Waypoint Launch Bearings-Only Long Detection Range")
+                    || curMode().equals("Waypoint Launch Bearings-Only Medium Detection Range")
+                    || curMode().equals("Waypoint Launch Bearings-Only Short Detection Range")) 
+                && getEntity().isSpaceborne()) {
             return true;
         }
         return false;
     }
     
+    /**
+     * Method that checks to see if our capital missile bay is in waypoint launch mode
+     * Only available in space games
+     * @return
+     */
+    public boolean isInWaypointLaunchMode() {
+        if ((curMode().equals("Waypoint Launch Bearings-Only Extreme Detection Range")
```

Comment:
Not sure I understand what's wrong with this.  I wrote the 'isinmode' methods to help me avoid magic strings in other classes.  I was unable to use them for 'setmodesformaptype'  because the 'isinmode' methods include a check that the entity is spaceborne. 

Do we really gain anything from tokenizing the text itself for use by the 'addmode' and 'removemode' commands?  No other weapons with modes work that way.

Reply:
Two reasons to use the defined tokens. 

1) The current mode is a user-visible string. 
2) You're actually using the value of the string to drive logic. So, you're comparing if the "current mode" is equal to [long string], and are using [long string] in multiple places. If the next guy working on the code goes to make a grammar adjustment or whatever to one of those places, they may not know to check the other places where you use that [long string], and they're going to have a bad time.

So, at the very least, you'll want to define "Waypoint Launch tra la la la" as string constants somewhere.

Reply:
Fair enough.  Weapon is probably the place to do this. That's where most of the mode strings appear to be introduced for the first time.